pkg/config/config.go

```diff
@@ -292,20 +292,21 @@ type Before struct {
 
 // S3 contains s3 config
 type S3 struct {
-	Region   string
-	Bucket   string
-	Folder   string
-	Profile  string
-	Endpoint string // used for minio for example
-	ACL      string
+	Region   string   `yaml:",omitempty"`
+	Bucket   string   `yaml:",omitempty"`
+	Folder   string   `yaml:",omitempty"`
+	Profile  string   `yaml:",omitempty"`
+	Endpoint string   `yaml:",omitempty"` // used for minio for example
+	ACL      string   `yaml:",omitempty"`
 	IDs      []string `yaml:"ids,omitempty"`
 }
 
 // Blob contains config for GO CDK blob
 type Blob struct {
-	Bucket   string
-	Provider string
-	Folder   string
+	Bucket   string   `yaml:",omitempty"`
+	Provider string   `yaml:",omitempty"`
+	Folder   string   `yaml:",omitempty"`
+	KMSKey   string   `yaml:",omitempty"`
```

Comment:
@caarlos0 I'm keen to test it, but I'm using `AES256` and not a custom KMS key. I had a look at the code but don't see a way to specify it.

Reply:
Also bumped into another issue. `blob` provider does not support the default authentication mechanisms provided by the AWS SDK, so IAM roles have no effect. That's a no-go for us at this stage, as our deployment environment does not have any long-living credentials available.

Reply:
got it... let me think about it some more ;)

Reply:
thanks for the review btw

Reply:
@caarlos0  Sorry for not being so active on this PR(vacation time). That's a really nice feature to have.
As I know,  S3/GCS/AZBLOB supports Server-side encryption just providing your key or respective KMS key reference. But go cloud is kind of blocker for this and really liked the way you wrapped it up by uploading encrypted data by our own. 👍 
I will test it as soon as possible.

@rafaelmagu 
> Also bumped into another issue. `blob` provider does not support the default authentication mechanisms provided by the AWS SDK, so IAM roles have no effect.

Are you talking about the IAM Roles for EC2 Instances? I have to check if there is a way to get access keys from such IAM role to make AWS calls.

Reply:
I guess [this](https://github.com/aws/aws-sdk-go/blob/master/aws/credentials/ec2rolecreds/ec2_role_provider.go#L86) is how AWS GO SDK reads credentials from EC2 instances. 
@caarlos0 do you think it makes sense to have something like this? We can define an order for retrieving credentials in our environment check function.

For instance:
- IAM Role
- Environment Variable.

Reply:
yeah I think it makes sense...

on the s3 repo, we read from role, env and ~/.aws/profile if I'm not mistaken...

if this is doable with gocloud too it would be awesome!

Reply:
> As I know, S3/GCS/AZBLOB supports Server-side encryption just providing your key or respective KMS key reference. But go cloud is kind of blocker for this and really liked the way you wrapped it up by uploading encrypted data by our own. 👍

thanks! Was the only way I found to do it...  haha

Reply:
> Sorry for not being so active on this PR(vacation time). Btw this is a really nice feature to have.

no problem at all! :)

Reply:
@caarlos0 
> yeah I think it makes sense...
> 
> on the s3 repo, we read from role, env and ~/.aws/profile if I'm not mistaken...
> 
> if this is doable with gocloud too it would be awesome!

I had a quick look in gocloud authentication. It's using AWS SDK which means it supports authentication mechanisms(IAM Roles, credentials file and ENV variables)by default.

Reason why it's **not** working now: 

As we discussed earlier, i was forcing the blob provider to use only environment variable as of now: 
`case "s3":
		return checkEnv("AWS_ACCESS_KEY", "AWS_SECRET_KEY", "AWS_REGION")`

[reference](https://github.com/goreleaser/goreleaser/blob/master/internal/pipe/blob/util.go#L16)

I just tried skipping this env variable check and used aws credentials file instead. It worked, did not test with IAM Role though, but I should work as well 👍 

We could skip this check but then we need to deal with errors from gocloud something like this:

`⨯ release failed after 8.66s error=Blob: failed to publish artifacts: failed to close Bucket writer: blob (code=Unknown): NoCredentialProviders: no valid providers in chain. Deprecated.
	For verbose messaging see aws.Config.CredentialsChainVerboseErrors`

**Worst in case of Azure**: if you don't specify AZURE_STORAGE_KEY gocloud won't complain and at the end it will throw Azure HTTP response for ResourceNotFound.

Any ideas, suggestion :) ?

Reply:
maybe just wrap the error with something like:

```go
errors.Wrapf(err, "failed to upload, check credentials and access to bucket %s", bucket)
```

?

Reply:
not very descriptive, but would in theory be correct 🤔

Reply:
As much as possible, the CDK should trust the chain provided by the AWS SDK. If it has support for that already, then this becomes a lot easier to test and covers more deployment scenarios.

If you can wrap the error but still surface the exception class to the user, that _should_ be enough context.

Reply:
@caarlos0 @rafaelmagu

Yeah. I will come up with new PR for this. 👍

Reply:
so, the way this is pr is implemented won't work, correct?

Reply:
afaik there is no option for that https://gocloud.dev/howto/secrets/crypt/

Reply:
So I managed to test this and it works with aliases, but no `AES256` support. For now it's good to go.

Reply:
awesome, will merge to master :)

eventually I'll need to figure out a way to cover this package with tests... several things are not covered... don't know how to do it yet though :(